src/cryptography/x509/general_name.py

```diff
@@ -123,7 +123,7 @@ def __ne__(self, other):
         return not self == other
 
     def __hash__(self):
-        return hash(self.value)
+        return hash((type(self), self.bytes_value))
```

Comment:
Soooooo, in a very limited, technical sense, these are wrong.

Only rule of hashing, equal objects must have the same hash. 

If you make your own subclass of one of these, it won't.

Let's just do the simple, obviously correct thing, and use only `bytes_value` in the hash, it'll be correct always, and I'm not going to lose a lot of sleep over hash collisions from objects with the same values but different types. If you put an IP address in a DNSName, either you work for a major CA, or you're very confused about what a DNSName is.